src/frontend/Typechecker.ml

```diff
@@ -1081,9 +1081,9 @@ let overlapping_lvalues lvals =
     | LIndexed (l, _), _ -> compare_no_indexing l lv2
     | _, LIndexed (l, _) -> compare_no_indexing lv1 l
     | LVariable id1, LVariable id2 -> String.compare id1.name id2.name
-    | LTupleProjection (lv1, idx1), LTupleProjection (lv2, idx2)
-      when idx1 = idx2 ->
-        compare_no_indexing lv1 lv2
+    | LTupleProjection (lv1, idx1), LTupleProjection (lv2, idx2) ->
+        let idx_comp = Int.compare idx1 idx2 in
+        if idx_comp = 0 then compare_no_indexing lv1 lv2 else idx_comp
```

Comment:
This was a "fun" one I realized after getting a build on 5.3.0, which is as previously implemented the sort here could actually be non-transitive, which led to a newly-optimized version of the `List.find_all_dups` function to miss some duplicates and allow an existing bad test to instead compile

Reply:
It's reassuring to know that our test coverage is wide enough to catch even such subtle bugs.

Reply:
That's certainly true -- it was painful to debug, but would have been much worse to not know about it at all